backend/schema/validate.go

```diff
@@ -38,15 +38,21 @@ var (
 //
 // This is useful for testing.
 func MustValidate(schema *Schema) *Schema {
-	clone, err := Validate(schema)
+	clone, err := ValidateWholeSchema(schema)
 	if err != nil {
 		panic(err)
 	}
 	return clone
 }
 
-// Validate clones, normalises and semantically valies a schema.
-func Validate(schema *Schema) (*Schema, error) {
+// ValidateWholeSchema clones, normalises and semantically validates a schema.
+func ValidateWholeSchema(schema *Schema) (*Schema, error) {
```

Comment:
ubernit: the "Whole" here seems redundant, but also isn't consistent with `ValidateModule()` - `schema.Validate()` is more idiomatic because it doesn't "stutter", but `schema.ValidateSchema()` could be preferable now because we now have a few different `Validate*()` functions